community/kernel/src/main/java/org/neo4j/kernel/impl/store/CommonAbstractStore.java

```diff
@@ -758,13 +766,16 @@ public final void visitStore( Visitor<CommonAbstractStore,RuntimeException> visi
      * can't trust that to be true. If we happen to have id generators open during recovery we delegate
      * {@link #freeId(long)} calls to {@link IdGenerator#freeId(long)} and since the id generator is most likely
      * out of date w/ regards to high id, it may very well blow up.
+     *
+     * This also marks the store as not OK. A call to {@link #makeStoreOk()} is needed once recovery is complete.
      */
     final void deleteIdGenerator()
     {
         if ( idGenerator != null )
         {
             idGenerator.delete();
             idGenerator = null;
+            setStoreNotOk( null );
```

Comment:
Could maybe use `new IllegalStateException( "IdGenerator is not initialized" )` as cause for store not being ok.